components/widgets/inputs/dropdown_input_hybrid.tsx

```diff
@@ -0,0 +1,253 @@
+import './dropdown_input_hybrid.scss';
+
+type ValueType = {
+    label: string | JSX.Element;
+    value: string;
+}
+
+type Props<T> = Omit<SelectProps<T>, 'onChange'> & {
+    value: T;
+    legend?: string;
+    error?: string;
+    onDropdownChange: (value: T) => void;
+    onInputChange: (event: React.ChangeEvent<HTMLInputElement>) => void;
+    placeholder: string;
+    className?: string;
+    name?: string;
+    exceptionToInput: string[];
+    width: number;
+    inputValue: string;
+    inputType?: string;
+    defaultValue: T;
+};
+
+const baseStyles = {
+    input: (provided: CSSProperties) => ({
+        ...provided,
+        color: 'var(--center-channel-color)',
+    }),
+    control: (provided: CSSProperties) => ({
+        ...provided,
+        border: 'none',
+        boxShadow: 'none',
+        padding: '0 2px',
+        cursor: 'pointer',
+    }),
+    indicatorSeparator: (provided: CSSProperties) => ({
+        ...provided,
+        display: 'none',
+    }),
+    menuPortal: (provided: CSSProperties) => ({
+        ...provided,
+        zIndex: 99999999,
+    }),
+};
+
+
+const IndicatorsContainer = (props: IndicatorContainerProps<ValueType>) => (
+    <div className='DropdownInput__indicatorsContainer'>
+        <components.IndicatorsContainer {...props}>
+            <i className='icon icon-chevron-down'/>
+        </components.IndicatorsContainer>
+    </div>
+);
+
+const Control = (props: ControlProps<ValueType>) => (
+    <div className='DropdownInput__controlContainer'>
+        <components.Control {...props}/>
+    </div>
+);
+
+const Option = (props: OptionProps<ValueType>) => (
+    <div
+        className={classNames('DropdownInput__option', {
+            selected: props.isSelected,
+            focused: props.isFocused,
+        })}
+    >
+        <components.Option {...props}/>
+    </div>
+);
+
+const DropdownInputHybrid = <T extends ValueType>(props: Props<T>) => {
+    const {
+        value,
+        placeholder,
+        className,
+        name,
+        legend, 
+        onDropdownChange,
+        onInputChange,
+        error,
+        exceptionToInput,
+        width,
+        inputValue,
+        inputType,
+        defaultValue,
+        ...otherProps
+    } = props;
+
+    const containerRef = useRef<HTMLInputElement>(null);
+    const inputRef = useRef<HTMLInputElement>(null);
+    const [inputFocused, setInputFocused] = useState(false);
+    const [focused, setFocused] = useState(false);
+    const [showInput, setShowInput] = useState(!!inputValue);
+
+    useEffect(() => {
+        if (showInput && !inputValue && inputRef.current) {
+            inputRef.current.focus();
+        }
+    }, [showInput]);
+
+    useEffect(() => {
+        if (!inputFocused) {
+            showTextInput(inputValue, false);
+        }
+    }, [inputValue]);
+
+    useEffect(() => {
+        if (!inputValue && !focused && !inputFocused) {
+            onDropdownChange(props.defaultValue);
```

Comment:
Below there's an if check for onDropdownChange, should the same apply here?
(and `defaultValue` is destructured above)
```suggestion
            onDropdownChange?.(defaultValue);
```